packages/types/src/joiner/index.ts

```diff
@@ -0,0 +1,31 @@
+export type JoinerRelationship = {
+  alias: string
+  foreignKey: string
+  primaryKey: string
+  serviceName: string
+  inverse?: boolean // In an inverted relationship the foreign key is on the other service and the primary key is on the current service
+}
+
+export interface JoinerServiceConfig {
+  serviceName: string
+  primaryKeys: string[]
+  relationships: JoinerRelationship[]
+}
+
+export interface JoinerArgument {
+  name: string
+  value?: any
+  field?: string
+}
+
+export interface RemoteJoinerQuery {
+  service: string
+  expands?: Array<{
+    property: string
+    fields: string[]
```

Comment:
I know that we discuss multiple times to have object instead of a collection of string. Though, I was thinking that we have the utils to trasnform from string to object and vice verca so I am fine to start that way and see later at the usage if we want to move to object instead for all the string collection

Reply:
In this place I think strings are just fine, as we're composing the final selected field names from the entities.
When we have a more robust interface to be used as the input for every methode `list` of a Module, then I agree we should come up with a better typing alternative as we discussed before.